src/python_testing/TC_RR_1_1.py

```diff
@@ -0,0 +1,432 @@
+
+        # Make sure all certificates are installed with maximal size
+        dev_ctrl.fabricAdmin.certificateAuthority.maximizeCertChains = True
+
+        # TODO: Do from PICS list. The reflection approach here what a real client would do,
+        #       and it respects that the test says "TH writes 4 entries per endpoint where LabelList is supported"
+        logging.info("Pre-condition: determine whether any endpoints have UserLabel cluster (ULABEL.S.A0000(LabelList))")
+        endpoints_with_user_label_list = await dev_ctrl.ReadAttribute(self.dut_node_id, [Clusters.UserLabel.Attributes.LabelList])
+        has_user_labels = len(endpoints_with_user_label_list) > 0
+        if has_user_labels:
+            logging.info("--> User label cluster present on endpoints %s" %
+                         ", ".join(["%d" % ep for ep in endpoints_with_user_label_list.keys()]))
+        else:
+            logging.info("--> User label cluster not present on any endpoitns")
+
+        # Generate list of all clients names
+        all_names = []
+        for fabric_idx in range(num_fabrics_to_commission):
+            for controller_idx in range(num_controllers_per_fabric):
+                all_names.append("RD%d%s" % (fabric_idx + 1, chr(ord('A') + controller_idx)))
+        logging.info("Client names that will be used: %s" % all_names)
+        client_list = []
+
+        # TODO: Shall we also verify SupportedFabrics attribute, and the CapabilityMinima attribute?
+        logging.info("Pre-conditions: validate CapabilityMinima.CaseSessionsPerFabric >= 3")
+
+        capability_minima = await self.read_single_attribute(dev_ctrl, node_id=self.dut_node_id, endpoint=0, attribute=Clusters.Basic.Attributes.CapabilityMinima)
+        asserts.assert_greater_equal(capability_minima.caseSessionsPerFabric, 3)
+
+        # Step 1: Commission 5 fabrics with maximized NOC chains
+        logging.info("Step 1: use existing fabric to configure new fabrics so that total is %d fabrics" %
+                     num_fabrics_to_commission)
+
+        # Generate Node IDs for subsequent controllers start at 200, follow 200, 300, ...
+        node_ids = [200 + (i * 100) for i in range(num_controllers_per_fabric - 1)]
+
+        # Prepare clients for first fabric, that includes the default controller
+        dev_ctrl.name = all_names.pop(0)
+        client_list.append(dev_ctrl)
+
+        if num_controllers_per_fabric > 1:
+            new_controllers = await CommissioningBuildingBlocks.CreateControllersOnFabric(fabricAdmin=dev_ctrl.fabricAdmin, adminDevCtrl=dev_ctrl, controllerNodeIds=node_ids, privilege=Clusters.AccessControl.Enums.Privilege.kAdminister, targetNodeId=self.dut_node_id)
+            for controller in new_controllers:
+                controller.name = all_names.pop(0)
+            client_list.extend(new_controllers)
+
+        # Prepare clients for subsequent fabrics
+        for i in range(num_fabrics_to_commission - 1):
+            admin_index = 2 + i
```

Comment:
Is there a specific reason this is different for each fabric? Can't it not all be the same?

Reply:
Just helped me track and I had tested this was in TC-SC-3.6. The test plan does not specify fabric ID to use.